### ISSUE TITLE
bug: prompt version specified on js sdk keeps retrieving the prompt that is in production

### ISSUE THREAD

**Issue by Heucles** [langfuse/langfuse#2847]
### Describe the bug

Basically I am trying to retrieve a specific version of a prompt based on the version number or a label, but even if I provide than to the `getPrompt` method, I will always get the prompt tagged with production, and if I leave all the versions without a production tag I get the error: `Error: No production-labeled prompt found with name ...`

### To reproduce

```js   
this.langfuseClient = new Langfuse({
      baseUrl: process.env.LLM_PROMPT_MANAGER_BASE_URL,
      secretKey: process.env.LLM_PROMPT_MANAGER_SECRET_KEY,
      publicKey: process.env.LLM_PROMPT_MANAGER_PUB_KEY,
    });

const prompt = await this.langfuseClient.getPrompt(promptName, x}); // x here is any int which has a version on langchain
    
```

### SDK and container versions

    "langfuse": "^3.16.2",
    "langfuse-core": "^3.16.2",
    "langfuse-langchain": "^3.16.2"

### Are you interested to contribute a fix for this bug?

Yes

**Comment by marcklingen** [comment]
thanks for sharing, just had a minute to add a test case for this here and will look into this: https://github.com/langfuse/langfuse-js/pull/338

**Comment by marcklingen** [comment]
Test passes successfully

**Comment by marcklingen** [comment]
Is the } after x a typo?

**Comment by Heucles** [comment]
Yes it is @marcklingen, sorry about that, was from an interation in which I've tried to get it based on a label through the options object

**Comment by marcklingen** [comment]
Does this mean this works as expected now and can be closed?

**Comment by Heucles** [comment]
No @marcklingen was mistype, but the error unfortunately is happening

**Comment by marcklingen** [comment]
Can you check if the credentials are set correctly?

As mentioned above, we now have an integration test that specifically checks this functionality, so it should work.

**Comment by Heucles** [comment]
@marcklingen sorry for the delayed response, but for me it happens, and about the credentials they are correct, maybe if I update the sdk or even langfuse itself it will fix it

**Comment by marcklingen** [comment]
Sorry I missed that you self-host Langfuse. Could you please provide the version number? We recently released a v2 prompts API, which is now in use. Generally, if you upgrade the SDKs, you also need to update the application container. The application container is backward compatible with old SDK versions, but not the other way around.
